### PR TITLE
Fix agent panic in case of malformed objects retrieved from the kvstore, and improve validation

### PR DESCRIPTION
**Pull request by giorio94**
This PR adds extra validation to prevent processing invalid node and service objects retrieved from the kvstore, which could trigger unexpected behavior.

```release-note
Fix agent panic in case malformed objects are retrieved from the kvstore, and improve validation
```

### CODE REVIEW

File: pkg/node/types/node.go [hunk 1]
```diff
@@ -615,7 +616,23 @@ func (n *Node) Unmarshal(_ string, data []byte) error {
...
+func (n *Node) validate() error {
+	// Skip the ClusterID check if it matches the local one, as we assume that
+	// it has already been validated, and to allow it to be zero.
+	if n.ClusterID != option.Config.ClusterID {
+		if err := cmtypes.ValidateClusterID(n.ClusterID); err != nil {
+			return err
+		}
+	}
+
+	return nil
```
Comment (pippolo84) [inline pkg/node/types/node.go:637]: Possible style improvement:

```suggestion
	if n.ClusterID == option.Config.ClusterID {
		return nil
	}

	return cmtypes.ValidateClusterID(n.ClusterID)
```

**Review by pippolo84 — APPROVED** [review]
LGTM! :rocket: 
I've left inline a non-blocking style improvement advice, feel free to do as you please.

File: pkg/node/types/node.go (hunk 1 above)
Comment (giorio94) [reply pkg/node/types/node.go:637]: Generally speaking, I agree that your proposal is cleaner. I'd personally prefer to preserve the current approach, though, as it simplifies the addition of further checks, and it is consistent with the services validation.

**Review by squeed — APPROVED** [review]

**Comment by giorio94** [comment]
The last push reverted a minor modification for better consistency with the previous implementation, with no functional differences.

**Review by ysksuzuki — APPROVED** [review]

**Pull request merged**